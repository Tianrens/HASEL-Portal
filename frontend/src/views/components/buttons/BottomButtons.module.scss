@import '../../../assets/responsiveDesignHelpers';
@import '../../../assets/colours';

.buttonContainer {
    width: 100%;
    display: flex;
    justify-content: space-between;
    & > div {
        display: flex;
    }
}

.divider {
    width: 100%;
    margin: 3rem 0;
}

@include responsive(tablet) {
    .buttonContainer,
    .buttonContainer > div {
        flex-direction: column;
    }
}
