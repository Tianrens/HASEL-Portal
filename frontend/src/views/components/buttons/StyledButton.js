import React from 'react';
import { Button } from '@material-ui/core';
import styles from './StyledButton.module.scss';

export function StyledButton({ color, size, icon, onClick, children, type }) {
    return (
        <Button
            variant='contained'
            onClick={onClick}
            startIcon={icon} // Must be a Material UI component
            className={`${styles.button} ${styles[color]}`}
            size={size}
            type={type}
        >
            {children}
        </Button>
    );
}
