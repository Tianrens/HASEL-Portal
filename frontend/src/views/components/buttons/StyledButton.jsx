import React from 'react';
import { Button } from '@mui/material';
import styles from './StyledButton.module.scss';

export function StyledButton({
    color,
    outline,
    size = 'large',
    icon,
    onClick,
    children,
    type,
    form,
    ...props
}) {
    return (
        <Button
            variant='contained'
            onClick={onClick}
            startIcon={icon} // Must be a Material UI component
            className={`${styles.button} ${styles[color]} ${outline ? styles.outline : ' '} ${
                size === 'small' ? styles.small : ' '
            }`}
            size={size}
            type={type}
            form={form}
            {...props}
        >
            {children}
        </Button>
    );
}
