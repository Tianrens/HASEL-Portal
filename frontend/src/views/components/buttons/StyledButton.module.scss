@import '../../../assets/colours.module';
@import '../../../assets/responsiveDesignHelpers';

@mixin theme($theme: $purple700) {
    background-color: $theme;
    border: 2px solid $theme;
    &:hover,
    &.outline {
        background-color: transparent;
        color: $theme;
    }
    &.outline:hover {
        background-color: $theme;
        color: white;
    }
}

.button {
    @include theme;

    border-radius: 32px;
    color: white;
    display: flex;
    font-family: NationalBold, sans-serif;
    font-weight: bold;
    margin: 0.5em;
    max-width: 100%;
    min-height: 3em;
    overflow: hidden;
    padding: 0.8em 2em;
    text-align: center;
    text-decoration: none;
    text-transform: none;
    text-overflow: ellipsis;
    transition: 0.3s;
    vertical-align: middle;
    white-space: nowrap;
}

.lighter {
    @include theme($theme: $purple500);
}

.green {
    @include theme($theme: $green);
}

.yellow {
    @include theme($theme: $yellow);
}

.red {
    @include theme($theme: $red);
}

@include responsive(tablet) {
    .small {
        padding: 0.6em 2em;
    }
}
