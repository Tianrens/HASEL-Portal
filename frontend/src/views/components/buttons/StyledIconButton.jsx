import React from 'react';
import { IconButton } from '@mui/material';
import styles from './StyledIconButton.module.scss';

export function StyledIconButton({ color, icon, onClick, disabled }) {
    return (
        <IconButton
            onClick={onClick}
            className={`${styles.iconButton} ${styles[color]} ${disabled ? styles.disabled : ''}`}
            disabled={disabled}
            size="large">
            {icon}
        </IconButton>
    );
}
