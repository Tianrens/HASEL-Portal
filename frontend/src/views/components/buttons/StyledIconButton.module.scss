@import './StyledButton.module';

.iconButton {
    @extend .button;
    border-radius: 100px;
    min-height: 1.5em;
    min-width: 1.5em;
    margin: 0.2em;
    padding: 0.2em 0.2em;
}

.disabled {
    border-color: $disabled;
}
