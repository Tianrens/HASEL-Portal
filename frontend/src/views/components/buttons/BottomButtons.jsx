import { React, useState } from 'react';
import Icon from '@mui/material/Icon';
import { StyledButton } from './StyledButton';
import { buttonContainer } from './BottomButtons.module.scss';
import ConfirmDialog from '../Modal/ConfirmDialog';

const BottomButtons = ({
    onDelete,
    deleteText,
    onDeny,
    denyText,
    onAccept,
    acceptText,
    acceptDisabled,
    deleteMessage,
    denyMessage,
    acceptMessage,
}) => {
    const [open, setOpen] = useState(false);
    const [message, setMessage] = useState('');
    const [onSubmit, setOnSubmit] = useState();

    const openModal = (msg, fn) => {
        setMessage(msg);
        setOnSubmit(() => fn);
        setOpen(true);
    };

    return (
        <>
            <div className={buttonContainer}>
                <div>
                    {onDelete && (
                        <StyledButton
                            color='red'
                            icon={<Icon>delete</Icon>}
                            type='submit'
                            onClick={() => openModal(deleteMessage, onDelete)}
                        >
                            {deleteText ?? 'Delete'}
                        </StyledButton>
                    )}
                </div>
                <div>
                    {onDeny && (
                        <StyledButton
                            color="lighter"
                            outline
                            icon={<Icon>close</Icon>}
                            type='submit'
                            onClick={() => openModal(denyMessage, onDeny)}
                        >
                            {denyText ?? 'Deny'}
                        </StyledButton>
                    )}
                    {onAccept && (
                        <StyledButton
                            icon={<Icon>done</Icon>}
                            type='submit'
                            onClick={() => openModal(acceptMessage, onAccept)}
                            disabled={acceptDisabled}
                        >
                            {acceptText ?? 'Accept'}
                        </StyledButton>
                    )}
                </div>
            </div>
            <ConfirmDialog
                message={message}
                open={open}
                onSubmit={onSubmit}
                handleClose={() => setOpen(false)}
            />
        </>
    );
};

export default BottomButtons;
