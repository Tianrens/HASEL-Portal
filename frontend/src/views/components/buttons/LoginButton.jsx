import { Icon } from '@mui/material';
import React from 'react';
import { firebase, auth } from '../../../firebase';
import { StyledButton } from './StyledButton';

async function login(callback, errorCallback) {
    const provider = new firebase.auth.GoogleAuthProvider();

    await auth.signInWithPopup(provider).then(() => {
        const userEmail = auth.currentUser.email;
        if (userEmail.endsWith('@aucklanduni.ac.nz') || userEmail.endsWith('@auckland.ac.nz')) {
            callback();
        } else {
            auth.signOut();
            errorCallback('Not using a University of Auckland email.');
        }
    });
}

/**
 * Button to log in. Upon successful login, calls the callback and saves the authentication details in the AppProvider.
 * Upon unsuccessful login, throws error.
 * @param {function callback specifies the function that is called after the login is successful}
 */
function LoginButton({ callback, errorCallback }) {
    return (
        <StyledButton
            icon={<Icon>login</Icon>}
            onClick={() => {
                login(callback, errorCallback);
            }}
        >
            Login
        </StyledButton>
    );
}

export default LoginButton;
