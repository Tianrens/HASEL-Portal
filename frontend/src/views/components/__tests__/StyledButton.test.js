import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import { StyledButton } from '../buttons/StyledButton';

test('StyledButton renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <StyledButton />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
