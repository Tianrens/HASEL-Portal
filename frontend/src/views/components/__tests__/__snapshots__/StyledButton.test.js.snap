// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton renders properly 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button undefined css-sghohy-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
/>
`;
